### DISCUSSION TITLE
Inventree on Plesk

### DISCUSSION THREAD

**Discussion by italianjon** [inventree/InvenTree#6065] [General]
Does anyone have a simple guide for installing Inventree to Plesk? I have been going around in circles all morning.

I get the output below and the first line implies I need to run the "docker compose" - no matter what I try I can not get it into the server environment of plesk environment. I am obviously missing something pretty simple - but I just can't see it

Thanks in advance

Failed to query label for 'label/inventree/stockitem/qr.html' - you should run 'invoke update' first!
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/base/base.py", line 219, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.10/site-packages/django/utils/asyncio.py", line 33, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/base/base.py", line 200, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.10/site-packages/django/utils/asyncio.py", line 33, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/mysql/base.py", line 234, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python3.10/site-packages/MySQLdb/__init__.py", line 121, in Connect
    return Connection(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/MySQLdb/connections.py", line 193, in __init__
    super().__init__(*args, **kwargs2)
MySQLdb.OperationalError: (2002, "Can't connect to local server through socket '/run/mysqld/mysqld.sock' (2)")

**Comment by SchrodingersGat** [comment]
Can you please provide a complete list of the steps you have followed, including the specific commands you are running

**Reply by italianjon** [reply]
Thanks for the fast answer.

I have the following steps:
1) Installed the Docker Extension into Plesk and imported the inventree/inventree image.
2) Setup a MySQL Database, with admin user linked to the subdomain I wish to use
3) Setup the variables as stated in the Documentation (EXT_VOLUME, DB_ENGINE, NAME, USER & PASS) as in the screenshot. But username and password redacted.

![Screenshot from 2023-12-10 13-54-54](https://github.com/inventree/InvenTree/assets/153430339/41ffd518-8469-40c6-b6a3-e13766f503bf)

4) Trying to start the Docker Container, gives the output above.
5) When I try and do the "docker compose run inventree-server invoke update" it is failing as compose isn't available, on install via a forced addition of a PPA, I can't seem to get the location of the downloaded docker image under plesk.

I would say I am far from expert, I would say I am probably intermediate in both dockers and plesk packages.

Thanks for support here.

**Comment by SchrodingersGat** [comment]
I am not familiar with running on plesk but it appears that you need to stage the containers separately. So, you will need another docker container which is hosting your database. You can use our `docker-compose.yml` file as a starting point, but the db container will need to be running (with a valid database) before the inventree container is started

**Comment by italianjon** [comment]
Thanks.

I've got a busy few days, but I'll try that before mid-week and post back.
Up to now it's only been tried with a non-docker MySQL database running
directly in the Plesk instance. I hadn't thought about the DB in a docker
container as well.

Until then, thanks again.
